src/main/host/syscall_condition.rs

```diff
@@ -1,43 +1,129 @@
 use crate::cshadow;
+use crate::host::descriptor::OpenFile;
 use crate::host::syscall::Trigger;
 
-/// Wrapper
+use std::marker::PhantomData;
+
+/// An immutable reference to a syscall condition.
 #[derive(Debug, PartialEq, Eq)]
-pub struct SysCallCondition {
+pub struct SysCallConditionRef<'a> {
     c_ptr: *mut cshadow::SysCallCondition,
+    _phantom: PhantomData<&'a ()>,
+}
+
+// do not define any mutable methods for this type
+impl<'a> SysCallConditionRef<'a> {
+    /// Borrows from a C pointer. i.e. doesn't increase the ref count, nor decrease the ref count
+    /// when dropped.
+    /// `ptr` must point to a valid object that will not be accessed by other threads
+    /// for the lifetime of this object.
+    pub fn borrow_from_c(ptr: *mut cshadow::SysCallCondition) -> Self {
```

Comment:
This seems like ought to be `unsafe`, since violating the contract above would be UB

Reply:
Made these functions `unsafe`. I left the `Thread::syscall_condition()` and `Thread::syscall_condition_mut()` functions as safe since I think if we have a `Thread` object, then we can already assume that their internal syscall condition isn't being modified from other threads.

Reply:
:+1: